zebra/zebra_vrf.c

```diff
@@ -101,6 +103,10 @@ static int zebra_vrf_new(struct vrf *vrf)
 	otable_init(&zvrf->other_tables);
 
 	router_id_init(zvrf);
+
+	if (!vrf_is_backend_netns())
+		vrf_table_id_add(&vrfs_by_table_id, zvrf);
```

Comment:
At this point in the code, the table_id is still DEFAULT.

The `table_id` is set later in `netlink_vrf_change()` in `zebra/if_netlink.c`.

Then `zebra_vrf_enable()` is called, so I would put the code to add it to the hash table there.

Right now, it just fails lookup every time because the `table_id` changes after its been put into the hash table.

the call to `vrf_table_id_add()` and `_del()` should be at the end of `zebra_vrf_enable()` and `zebra_vrf_disable()` probably.

```
diff --git a/zebra/zebra_vrf.c b/zebra/zebra_vrf.c
index 0ee23774d..e0fc86f9d 100644
--- a/zebra/zebra_vrf.c
+++ b/zebra/zebra_vrf.c
@@ -104,9 +104,6 @@ static int zebra_vrf_new(struct vrf *vrf)
 
        router_id_init(zvrf);
 
-       if (!vrf_is_backend_netns())
-               vrf_table_id_add(&vrfs_by_table_id, zvrf);
-
        return 0;
 }
 
@@ -150,6 +147,9 @@ static int zebra_vrf_enable(struct vrf *vrf)
                zvrf->import_check_table[afi] = table;
        }
 
+       if (!vrf_is_backend_netns())
+               vrf_table_id_add(&vrfs_by_table_id, zvrf);
+
        /* Kick off any VxLAN-EVPN processing. */
        zebra_vxlan_vrf_enable(zvrf);
 
@@ -241,6 +241,10 @@ static int zebra_vrf_disable(struct vrf *vrf)
                }
        }
 
+       if (!vrf_is_backend_netns())
+               vrf_table_id_del(&vrfs_by_table_id, zvrf);
+
+
        return 0;
 }
 
@@ -258,9 +262,6 @@ static int zebra_vrf_delete(struct vrf *vrf)
                zlog_debug("VRF %s id %u deleted", zvrf_name(zvrf),
                           zvrf_id(zvrf));
 
-       if (!vrf_is_backend_netns())
-               vrf_table_id_del(&vrfs_by_table_id, zvrf);
-
        /* clean-up work queues */
        for (i = 0; i < MQ_SIZE; i++) {
                struct listnode *lnode, *nnode;
```